app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -623,8 +624,24 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 refreshCurrentView(false);
                 break;
         }
+        mFormController.setPendingCalloutFormIndex(null);
     }
-    
+
+    private IntentWidget tryToGetIntentWidget() {
+        IntentWidget bestMatch = null;
+
+        //Ugh, copied from the odkview mostly, that's stupid
+        for(QuestionWidget q : ((ODKView)mCurrentView).getWidgets()) {
+            //Figure out if we have a pending intent widget
+            if (q instanceof IntentWidget) {
+                if(((IntentWidget) q).isWaitingForBinaryData() || bestMatch == null) {
```

Comment:
I see, you mean reimplement this function as

```return mFormController.getPendingCalloutFormIndex().whateverTheGetCurrentWidgetFunctionIsCalled()```

right? I wasn't certain that was possible, but your hinting that it is gives me courage so I'll give it a shot.